From: Richard Acayan <mailingradian@gmail.com>
Date: Tue, 26 Jul 2022 19:06:42 -0400
Subject: [PATCH 108/108] power: qcom_fg: silence -EPROBE_DEFER error

Sometimes, the devm_power_supply_register function will return
ERR_PTR(-EPROBE_DEFER) to make the driver subsystem probe the fuel guage
later. This is not an error and do not report it as such.

Signed-off-by: Richard Acayan <mailingradian@gmail.com>
---
 drivers/power/supply/qcom_fg.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/power/supply/qcom_fg.c b/drivers/power/supply/qcom_fg.c
index 625a625..2d4275c 100644
--- a/drivers/power/supply/qcom_fg.c
+++ b/drivers/power/supply/qcom_fg.c
@@ -1157,7 +1157,8 @@ static int qcom_fg_probe(struct platform_device *pdev)
 	chip->batt_psy = devm_power_supply_register(chip->dev,
 			&batt_psy_desc, &supply_config);
 	if (IS_ERR(chip->batt_psy)) {
-		dev_err(&pdev->dev, "Failed to register battery\n");
+		if (PTR_ERR(chip->batt_psy) != -EPROBE_DEFER)
+			dev_err(&pdev->dev, "Failed to register battery\n");
 		return PTR_ERR(chip->batt_psy);
 	}
 
